examples/gcp_bigquery_big_lake_table/main.tf

```diff
@@ -37,28 +42,77 @@ module "bigquery_dataset" {
+  connection_id = "big_lake_connection"
+  location      = local.CONST_GOOGLE_REGION_JAKARTA
+}
+
 // creating the BigQuery Big Lake table
 module "big_lake_table" {
   source = "../../modules/gcp_bigquery_big_lake_table"
 
   // Let the table detect schema automatically
   autodetect = true
   // Connection id to Big Lake table
-  connection_id = "big_lake_connection"
+  connection_id = module.big_lake_connection.id
   // dataset id that this table will be created in
-  dataset_id = "big_lake_poc"
+  dataset_id = module.bigquery_dataset.id
   // protect terraform from deleting the resource, set to false in this example because the test will need to be able to destroy it
   deletion_protection = false
   // description of the table
   description = "table descriptions"
-  // Hive partition mode for detect file pattern
+
   hive_partitioning_mode = "AUTO"
-  // dataset id
-  hive_source_uri_prefix = "hive_source_uri"
+  hive_source_uri_prefix = "gs://${google_storage_bucket.big_lake_data_source.name}/"
   // name of this table, the table name will be name with run number, but the friendly name will be the same with what we set here
   name = "big_lake_table"
   // Format of source NEWLINE_DELIMITED_JSON, AVRO, PARQUET
-  source_format = "NEWLINE_DELIMITED_JSON"
+  source_format = local.CONST_BIGQUERY_SOURCE_FORMAT_PARQUET
   // source uri that let Big Lake table read the external data from GCS
-  source_uris = ["source_uri_pattern"]
+  source_uris = ["gs://${google_storage_bucket.big_lake_data_source.name}/${google_storage_bucket_object.dummy_parquet_file.name}"]
+
+
+  dataset_kms_key_name = module.bigquery_dataset.customer_managed_key_id
+
+  depends_on = [
+    google_storage_bucket_object.dummy_parquet_file,
+    google_storage_bucket_iam_member.big_lake_connection_gcs_binding
+  ]
+}
+
+resource "google_storage_bucket" "big_lake_data_source" {
+  #checkov:skip=CKV_GCP_114:This is an ephemeral example not meant for real-world usage.
+  #checkov:skip=CKV_GCP_29:This is an ephemeral example not meant for real-world usage.
+  #checkov:skip=CKV_GCP_62:This is an ephemeral example not meant for real-world usage.
+  #checkov:skip=CKV_GCP_78:This is an ephemeral example not meant for real-world usage.
+
+  location = local.CONST_GOOGLE_REGION_JAKARTA
+  name     = "big_lake_data_source-${random_id.big_lake_data_source_random_id.hex}"
+}
+
+resource "random_id" "big_lake_data_source_random_id" {
+  byte_length = 4
+}
+
+resource "google_storage_bucket_iam_member" "big_lake_connection_gcs_binding" {
+  bucket = google_storage_bucket.big_lake_data_source.id
+  member = "serviceAccount:${module.big_lake_connection.service_account_id}"
+  role   = "roles/storage.objectViewer"
+}
+
+locals {
+  file_extensions = {
+    AVRO                   = "avro",
+    CSV                    = "csv",
+    NEWLINE_DELIMITED_JSON = "json",
+    PARQUET                = "parquet",
+  }
+}
+
+resource "google_storage_bucket_object" "dummy_parquet_file" {
```

Comment:
After giving this more thought, I think placing empty files on GCS does not actually solve the deployment issues with the [data warehouse](https://github.com/honestbank/data-warehouse) repo. These files should have the same schema as the actual data coming in later, otherwise the actual data files could not be read by BigQuery.

On the other hand, the deployment errors that we have (see [this run](https://app.terraform.io/app/honestbank/workspaces/data-warehouse-qa/runs/run-8rxGGUih7Xd4iVaK) for example) are coming from BigQuery [schema auto detection](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#create_external_table_statement). So if we define the schemas in the code, nothing will be scanned by BigQuery when creating the external tables.

To fix the Terraform deployment issues, I think we have to define table schemas explicitly and deprecate/prevent using external tables without schema definition.